src/camera_positioner.cpp

```diff
@@ -59,12 +59,12 @@ class CameraPositioner {
       }
    }
 
-   void callback(const apriltags_ros::AprilTagDetectionArray& msg){
+   void callback(const apriltags2_ros::AprilTagDetectionArray& msg){
       // if we got a valid tag detection, update world_camera_transform
       for (int i=0; i< msg.detections.size(); i++) {
-        if(msg.detections[i].id == 0){
+        if(msg.detections[i].id[0] == 0){
```

Comment:
Even if this "can never happen", please check whether `id.size() > 0`.

Reply:
Fair enough! I wrapped wrapped all that cases by if conditions. Negative cases (should really never happen) are then handled as if no detection was received.